File: internal/ui/commands/compress.go
package commands

func registerCompress(r *Registry) {
	r.Register(&Command{Name: "compress", Description: "Compress conversation history", Kind: Handler,
		Handler: func(args []string, ctx Context) bool {
			if ctx.CompressFn != nil {
				ctx.CompressFn()
			}
			return true
		},
	})
}

File: src/ui/commands/builtin/compress.ts
import type { CommandHandler } from "../registry.js";

export const compressCommand: CommandHandler = {
  name: "compress",
  description: "Compress conversation context",
  handler: async (_args, ctx): Promise<void> => {
    await ctx.contextManager.compress();
    ctx.sessionManager.reportStatus({
      role: "status",
      content: "(Compression complete)",
    });
  },
};
